File: js/tests/stringToIntegerViceVersa.test.js
const { toInt, toStr } = require("../stringToIntegerViceVersa");

describe("toInt", () => {
  describe("Convert string to integer", () => {
    const data = [
      ["37", 37],
      ["113", 113],
      ["5", 5],
      ["5231", 5231]
    ];

    test.each(data)("toInt(%p) = %i", (str, expected) => {
      let actual = toInt(str);

      expect(actual).toEqual(expected);
    });
  });
});

describe("toStr", () => {
  describe("Convert integer to string", () => {
    const data = [
      [37, "37"],
      [113, "113"],
      [5, "5"],
      [5231, "5231"]
    ];

    test.each(data)("toStr(%i) = %p", (num, expected) => {
      let actual = toStr(num);

      expect(actual).toEqual(expected);
    });
  });
});

File: CSharp/Tests/StringToIntegerViceVersaTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class StringToIntegerViceVersaTest
    {
        [Theory]
        [InlineData("37", 37)]
        [InlineData("113", 113)]
        [InlineData("5", 5)]
        [InlineData("5231", 5231)]
        public void ToInt_StringInNumberFormat_ReturnIntFormatOfString(string str, int expected)
        {
            var actual = StringToIntegerViceVersa.ToInt(str);

            Assert.Equal(expected, actual);
        }

        [Theory]
        [InlineData(37, "37")]
        [InlineData(113, "113")]
        [InlineData(5, "5")]
        [InlineData(5231, "5231")]
        public void ToStr_NumberValue_ReturnStringFormatOfNumber(int num, string expected)
        {
            var actual = StringToIntegerViceVersa.ToStr(num);

            Assert.Equal(expected, actual);
        }
    }
}